### PR TITLE
Compilation of libodil on Windows (tested for VS 2015)

### PR DESCRIPTION
**Pull request by cguebert**
Modifications for CMake:
- Finding all DCMTK libraries
- Adding dependency to netapi32
- Setting the correct version of _WIN32_WINNT
- Grouping projects in the VS solution, and recreating the folders hierarchy for source files
- Linking libodil as a shared library

Bug fixes:
- Removed asking for a pointer in an empty vector (always crashes with VS).

Compilation fixes:
- VS could not deduce the correct templates for the new element getters in DataSet. Instead of writing the correct template manually, rewrote the methods to be more concise.

### CODE REVIEW

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -18,10 +18,13 @@ foreach(example_file ${examples})
     get_filename_component(example ${example_file} NAME_WE)
     add_executable(${example} ${example_file})
     target_link_libraries(${example} libodil)
+    install(TARGETS ${example} RUNTIME DESTINATION bin)
```
Comment (lamyj) [inline examples/CMakeLists.txt]: I'd rather keep the examples out of the installed resources: for the command-line interaction, we now have an full [application](https://github.com/lamyj/odil/tree/master/applications/odil)

File: src/odil/Reader.cpp [hunk 2]
```diff
@@ -59,6 +59,8 @@ type value; \
 
 std::string read_string(std::istream & stream, unsigned int size)
 {
+    if(size == 0)
```
Comment (lamyj) [inline src/odil/Reader.cpp]: Please avoid blocks without braces: the [goto fail](https://www.imperialviolet.org/2014/02/22/applebug.html) bug is still in many memories 😄

File: src/odil/Reader.cpp [hunk 3]
```diff
@@ -469,7 +471,9 @@ Reader::Visitor
 ::operator()(Value::Binary & value) const
 {
     auto const vl = this->read_length();
-    if(vl == 0xffffffff)
+    if(vl == 0)
```
Comment (lamyj) [inline src/odil/Reader.cpp]: Please avoid blocks without braces: the [goto fail](https://www.imperialviolet.org/2014/02/22/applebug.html) bug is still in many memories 😄

**Comment by lamyj** [comment]
Welcome to Odil and thanks for the pull request! I added a few minor comments to your commits: could you make sure my suggestions don't mess up the Windows compilation (especially the one regarding not installing the examples). Once you confirm, I'll merge.

File: examples/CMakeLists.txt (hunk 1 above)
Comment (cguebert) [inline examples/CMakeLists.txt]: Is it ok if I add an option in CMake to install the examples ? (Off by default)

**Comment by cguebert** [comment]
It seems the automatic builds are failing, but I cannot see exactly why. My modifications of FindDCMTK maybe?

**Comment by lamyj** [comment]
Coveralls does not really complain (less than half a percent in coverage change is not going to bother me), and Appveyor complains because my appveyor.yml file is broken: as long as Travis is happy, I'll merge.

Thanks!

**Pull request merged**